bot/cogs/moderation/infractions.py

```diff
@@ -235,8 +235,32 @@ async def apply_ban(self, ctx: Context, user: UserSnowflake, reason: str, **kwar
 
         Will also remove the banned user from the Big Brother watch list if applicable.
         """
-        if await utils.has_active_infraction(ctx, user, "ban"):
-            return
+        # In the case of a permanent ban, we don't need get_active_infractions to tell us if one is active
+        send_msg = "expires_at" in kwargs
+        active_infraction = await utils.get_active_infractions(ctx, user, "ban", send_msg)
+
+        if active_infraction:
+            log.trace("Active infractions found.")
+            if (
+                active_infraction.get('expires_at') is not None
+                and kwargs.get('expires_at') is None
```

Comment:
The kwargs check is present for both the `if` and the `elif`. You can move it out and negate it to exit early. This will also simplify the if statements:

```py
if kwargs.get('expires_at') is not None:
    # log something maybe
    return

if active_infraction.get('expires_at') is None:
    ...
else:
    ...
```

Reply:
Addressed in most recent commit